ArduSub/AP_Arming_Sub.cpp

```diff
@@ -12,11 +12,45 @@ bool AP_Arming_Sub::rc_calibration_checks(bool display_failure)
     return rc_checks_copter_sub(display_failure, channels);
 }
 
+bool AP_Arming_Sub::has_disarm_function () {
+    bool has_shift_function = false;
+    // make sure the craft has a disarm button assigned before it is armed
+    // check all the standard btn functions
+    for (uint8_t i = 0; i < 16; i++) {
+        switch (sub.get_button(i)->function(false)) {
+            case JSButton::k_shift :
+                has_shift_function = true;
+                break;
+            case JSButton::k_arm_toggle :
+                return true;
+            case JSButton::k_disarm :
+                return true;
+        }
+    }
+
+    // check all the shift functions if there's shift assigned
+    if (has_shift_function) {
```

Comment:
Could early-return here to save an indent level.  But really, just nitpicking here.

Reply:
im sorry, do you mean like this?
```
if (!has_shift_function) { return false; }
.
/// for and swich statements
.
```

Reply:
Yes.  But please note that I did hit "Approve", so this is really just minor, annoying stuff :-)

n.b. we'd actually want:

```
if (!has_shift_function) {
    return;
}
```